1.segment_images/scripts/0.segment_nuclei_organoids.py

```diff
@@ -0,0 +1,191 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# This notebook focuses on trying to find a way to segment cells within organoids properly.
+# The end goals is to segment cell and extract morphology features from cellprofiler.
+# These masks must be imported into cellprofiler to extract features.
+
+# ## import libraries
+
+# In[1]:
+
+
+import argparse
+import pathlib
+
+import matplotlib.pyplot as plt
+
+# Import dependencies
+import numpy as np
+import skimage
+import tifffile
+from cellpose import core, models
+from skimage import io
+
+# check if in a jupyter notebook
+try:
+    cfg = get_ipython().config
+    in_notebook = True
+except NameError:
+    in_notebook = False
+
+
+# ## parse args and set paths
+
+# In[2]:
+
+
+if not in_notebook:
+    print("Running as script")
+    # set up arg parser
+    parser = argparse.ArgumentParser(description="Segment the nuclei of a tiff image")
+
+    parser.add_argument(
+        "--input_dir",
+        type=str,
+        help="Path to the input directory containing the tiff images",
+    )
+    parser.add_argument(
+        "--window_size", type=int, help="Size of the window to use for the segmentation"
+    )
+    parser.add_argument(
+        "--clip_limit",
+        type=float,
+        help="Clip limit for the adaptive histogram equalization",
+    )
+
+    args = parser.parse_args()
+    window_size = args.window_size
+    clip_limit = args.clip_limit
+    input_dir = pathlib.Path(args.input_dir).resolve(strict=True)
+else:
+    print("Running in a notebook")
+    input_dir = pathlib.Path("../examples/raw_z_input/").resolve(strict=True)
+    window_size = 3
+    clip_limit = 0.05
```

Comment:
Recommend adding code comments here to define these variables more clearly. I know what window_size is based on being internally involved in the project but I don't know what clip_limit means.

Reply:
I have it defined in the argparse section too